src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/catalog.cpp

```diff
@@ -193,20 +193,19 @@ HRESULT WinRTGetMetadataFile(
     {
         return E_NOTIMPL;
     }
-
-    Microsoft::WRL::Wrappers::HString substring;
-    if (SUCCEEDED(WindowsSubstringWithSpecifiedLength(name, 0, 7, substring.GetAddressOf())))
+    wil::unique_hstring substring;
+    if (SUCCEEDED(WindowsSubstringWithSpecifiedLength(name, 0, 7, &substring)))
```

Comment:
Probably allocates a new string. Since the length is known, why not just copy it into a local wchar_t[8], and then you can call `CompareStringOrdinal` on the raw strings instead of `WindowsCompareStringOrdinal`.